### PR TITLE
updating computed compliance capabilities

### PR DESCRIPTION
**Pull request by sunto**
## Description

Automated compliance messages show up with text (PID, Plan Number, etc.)

## What type of PR is this? (check all applicable)

- [x ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## Related Tickets & Documents

(https://hous-bssb.atlassian.net/browse/BPHH-653)

## Steps to QA

Create a new application, it should set up PID and plan Number default values and show what the auto-compliance ofund as ad efault value.

## General Checklist

- [ x] ✅ Provide tests for your changes where relevant
- [ x] 📝 Use descriptive commit messages
- [ x] 📗 Update any related documentation and include any relevant screenshots
- [ ] 📱 For UI-related tasks, ensure responsive design is implemented across multiple screen size
- [ ] 🗂️ Move your relevant story/task to the _Ready for Code Review_ state

### CODE REVIEW

File: app/models/concerns/automated_compliance_utils.rb [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
-  #placheholder, will be removed once lookup_props is refactored
-  def published_requirement_template
-    RequirementTemplate.find_by(activity: activity, permit_type: permit_type)
-  end
-
-  def requirements_lookups
-    published_requirement_template&.lookup_props || {}
+  def requirements_lookups #a flattened list of requiements with the right information
+    # template_version&.lookup_props
+    template_version&.requirement_template&.lookup_props
```
Comment (taw27) [inline app/models/concerns/automated_compliance_utils.rb:6]: should we rely on the requirement_template? The current draft `requirement_template` state might be vastly different from what's there in the currently published `template_version` e.g. Removed and added a bunch of requirements that isn't in the current template version

File: app/services/integrations/ltsa_parcel_map_bc.rb [hunk 2]
```diff
@@ -26,7 +23,7 @@ def historic_site_by_pid(pid:)
     query = "where=HISTORIC_SITE_IND='Y' AND PARCEL_DESCRIPTION='#{pid}'&returnGeometry=true&outFields=*"
     response = @client.get("#{ENV["GEO_LTSA_PARCELMAP_REST_URL"]}#{HISTORIC_SERVICE}/query?f=json&#{query}")
     #assuem if there is a parcel description match not to use ltsa geometry matching
-    return response if response.success? && response.body.dig("features").length > 0
+    return parse_attributes_from_response(response) if response.success? && response.body.dig("features").length > 0
```
Comment (taw27) [inline app/services/integrations/ltsa_parcel_map_bc.rb:26]: what happens if features is nil

File: app/models/concerns/automated_compliance_utils.rb (hunk 1 above)
Comment (sunto) [reply app/models/concerns/automated_compliance_utils.rb:6]: yes..I think we have separate card to remove thise lookup_props from the requirment template.  https://hous-bssb.atlassian.net/browse/BPHH-846

File: app/services/integrations/ltsa_parcel_map_bc.rb (hunk 2 above)
Comment (sunto) [reply app/services/integrations/ltsa_parcel_map_bc.rb:26]: if the pid fials to be found (returns nil), it goes onto look fo rthe geometry of the pid, then suppies that geometry for a intersection with historic sites to see if there is any overlap

**Review by taw27 — APPROVED** [review]

**Pull request merged**